lib/ohai/plugins/aix/network.rb

```diff
@@ -56,20 +56,19 @@
       network[:default_interface] = default_line[5]
     end
 
-    # Splits the ifconfig output to 1 line per interface
-    if_so = shell_out("ifconfig -a").stdout
-    if_so.gsub(/\n(\w+\d+)/, '___\1').split("___").each do |intraface|
-      splat = intraface.split(":")
-      interface = splat[0]
-      line = splat[1..-1][0]
-      iface[interface] = Mash.new
-      iface[interface][:state] = (line.include?("<UP,") ? "up" : "down")
+    # Splits the ifconfig output into arrays of interface strings
+    shell_out("ifconfig -a").stdout.split(/\n(?=\w)/).each do |int_lines|
+      int_name, int_data = int_lines.split(":", 2)
 
-      intraface.each_line do |lin|
+      ifaces[int_name] = Mash.new
+      ifaces[int_name][:addresses] ||= Mash.new
+      ifaces[int_name][:state] = (int_data.include?("<UP,") ? "up" : "down")
+
+      int_data.each_line do |lin|
```

Comment:
While we're on the subject of bad variable names, could this be `line`?

Reply:
Fixed